src/meson.build

```diff
@@ -0,0 +1,33 @@
+modbus_version_h = configure_file(input : 'modbus-version.h.in',
+                                  output : 'modbus-version.h',
+                                  configuration : ver_conf)
+
+sources = files([
+    'modbus.c',
+    'modbus-data.c',
+    'modbus-rtu.c',
+    'modbus-tcp.c',
+])
+
+headers = [
+    'modbus.h',
+    modbus_version_h,
+    'modbus-rtu.h',
+    'modbus-tcp.h',
+]
+install_headers(headers, subdir : 'modbus')
+
+lt_current = 6
+lt_revision = 0
+lt_age = 1
+lt_version = '@0@.@1@.@2@'.format(lt_current, lt_age, lt_revision)
+
+libmodbus = shared_library('modbus', sources,
+                           version: lt_version,
+                           soversion: lt_current,
```

Comment:
If you drop the soversion definition, Meson will default to using the first component of the version, which is usually what people want (and is compatible with autotools).

Reply:
I feel like I'm missing understanding something... What's the point of doing `(C - A).(A).(R)` if `(C)` never gets used on its own? Why not just make `C=5` and just do `(C).(A).(R)`?